tests/test_methods_and_attributes.py

```diff
@@ -474,3 +474,13 @@ def test_custom_caster_destruction():
 
     # Make sure we still only have the original object (from ..._no_destroy()) alive:
     assert cstats.alive() == 1
+
+
+def test_overload_order_normal():
+    'Check to see if the normal overload order (first defined) works'
+    assert m.OverloadOrder("this").value == 1
+
+
+def test_overload_order_priority():
+    'Check to see if the prepend overload order works'
+    assert m.OverloadOrder(0).value == 4
```

Comment:
In addition to checking that the right overload was selected, this should also check that the generated docstring for the overloaded method is ordered correct (and not missing anything), and (similarly) that the exception message for invalid arguments is correct.